x-pack/plugins/observability_solution/observability_ai_assistant/scripts/evaluation/kibana_client.ts

```diff
@@ -201,6 +213,61 @@ export class KibanaClient {
       unregister: () => void;
     }> = [];
 
+    function serializeAndHandleRetryableErrors<
+      T extends StreamingChatResponseEvent
+    >(): OperatorFunction<Buffer, Exclude<T, ChatCompletionErrorEvent>> {
+      return (source$) => {
+        const processed$ = source$.pipe(
+          concatMap((buffer: Buffer) =>
+            buffer
+              .toString('utf-8')
+              .split('\n')
+              .map((line) => line.trim())
+              .filter(Boolean)
+              .map((line) => JSON.parse(line) as T | BufferFlushEvent)
+          ),
+          throwSerializedChatCompletionErrors(),
+          retry({
```

Comment:
Retry, but only when it's an internal error or an Axios error.

Reply:
Do we have examples of the errors returned here ? Wondering if 1 retry is enough